subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/SelectorState.java

```diff
@@ -303,7 +302,7 @@ private ComponentSelector selectorWithDesugaredAttributes(ComponentSelector sele
     }
 
     private static boolean isForced(DependencyState dependencyState) {
-        if (dependencyState.getRuleDescriptor() != null && dependencyState.getRuleDescriptor().getCause() == ComponentSelectionCause.FORCED) {
+        if (dependencyState.getRuleDescriptor() != null && dependencyState.getRuleDescriptor().isForceAlignment()) {
```

Comment:
:o: I would say that alignment is a side effect of force. So maybe it would be more correct to have something like `isEquivalentToForce()`.

Reply:
I mean `ComponentSelectionDescriptorInternal`

Reply:
Got it, updating